src/stirling/source_connectors/socket_tracer/mux_trace_bpf_test.cc

```diff
@@ -0,0 +1,176 @@
+#include "src/stirling/source_connectors/socket_tracer/testing/socket_trace_bpf_test_fixture.h"
+#include "src/stirling/testing/common.h"
+
+namespace px {
+namespace stirling {
+
+namespace mux = protocols::mux;
+
+using ::px::stirling::testing::ColWrapperSizeIs;
+using ::px::stirling::testing::FindRecordIdxMatchesPID;
+using ::px::stirling::testing::FindRecordsMatchingPID;
+using ::px::stirling::testing::SocketTraceBPFTest;
+using ::testing::AllOf;
+using ::testing::UnorderedElementsAre;
+using ::testing::UnorderedElementsAreArray;
+
+using ::testing::Each;
+using ::testing::Field;
+using ::testing::MatchesRegex;
+
+class MuxTraceTest : public SocketTraceBPFTest</* TClientSideTracing */ true> {
+ protected:
+  MuxTraceTest() {
+    // The container runner will make sure it is in the ready state before unblocking.
+    // Stirling will run after this unblocks, as part of SocketTraceBPFTest SetUp().
+
+    // Run the thriftmux server and then the thriftmux client container
+    PL_CHECK_OK(server_.Run(std::chrono::seconds{60}));
+  }
+
+  std::string thriftmux_client_output = "StringString";
+
+  std::string classpath =
+      "@/app/px/src/stirling/source_connectors/socket_tracer/testing/containers/thriftmux/"
+      "server_image.classpath";
+
+  StatusOr<int32_t> RunThriftMuxClient() {
+    std::string cmd =
+        absl::StrFormat("docker exec %s /usr/bin/java -cp %s Client & echo $! && wait",
+                        server_.container_name(), classpath);
+    PL_ASSIGN_OR_RETURN(std::string out, px::Exec(cmd));
+
+    LOG(INFO) << absl::StrFormat("thriftmux client command output: '%s'", out);
+
+    std::vector<std::string> tokens = absl::StrSplit(out, "\n");
+
+    int32_t client_pid;
+    if (!absl::SimpleAtoi(tokens[0], &client_pid)) {
+      return error::Internal("Could not extract PID.");
+    }
+
+    LOG(INFO) << absl::StrFormat("Client PID: %d", client_pid);
+
+    if (tokens[1] != thriftmux_client_output) {
+      return error::Internal(
+          absl::StrFormat("Expected output from thriftmux to be '%s', received '%s' instead",
+                          thriftmux_client_output, tokens[1]));
+    }
+    return client_pid;
+  }
+
+  ::px::stirling::testing::ThriftMuxServerContainer server_;
+};
+
+std::vector<mux::Record> ToRecordVector(const types::ColumnWrapperRecordBatch& rb,
+                                        const std::vector<size_t>& indices) {
+  std::vector<mux::Record> result;
+
+  for (const auto& idx : indices) {
+    mux::Record r;
+    uint24_t tag = static_cast<uint24_t>(rb[kMuxTagIdx]->Get<types::Int64Value>(idx).val);
+    r.req.tag = tag;
+    r.req.type = static_cast<int8_t>(rb[kMuxReqTypeIdx]->Get<types::Int64Value>(idx).val);
+    r.resp.tag = tag;
+    r.resp.type = static_cast<int8_t>(rb[kMuxRespTypeIdx]->Get<types::Int64Value>(idx).val);
+    result.push_back(r);
+  }
+  return result;
+}
+
+mux::Record GetRecordWithTagAndType(uint24_t tag, mux::Type req_type, mux::Type resp_type) {
```

Comment:
nit/naming: `GetRecordWithTagAndType` gets confusing with `GetTargetRecords` where "get" implies getting out of the records. Can we rename to just  `RecordWithTagAndType`

Reply:
Great point, updated.